editor/src/components/user-bar.tsx

```diff
@@ -0,0 +1,163 @@
+import { ClientSideSuspense } from '@liveblocks/react'
+import React from 'react'
+import { useOthers, useSelf, useStatus } from '../../liveblocks.config'
+import { getUserPicture, isLoggedIn } from '../common/user'
+import type { MultiplayerColor } from '../core/shared/multiplayer'
+import {
+  multiplayerColorFromIndex,
+  multiplayerInitialsFromName,
+  normalizeMultiplayerName,
+  normalizeOthersList,
+} from '../core/shared/multiplayer'
+import { Avatar, Tooltip, useColorTheme } from '../uuiui'
+import { Substores, useEditorState } from './editor/store/store-hook'
+
+const MAX_VISIBLE_OTHER_PLAYERS = 4
+
+export const UserBar = React.memo(() => {
+  const loginState = useEditorState(
+    Substores.userState,
+    (store) => store.userState.loginState,
+    'UserBar loginState',
+  )
+
+  const roomStatus = useStatus()
+
+  if (!isLoggedIn(loginState)) {
+    return null
+  } else if (roomStatus !== 'connected') {
+    return <SinglePlayerUserBar />
+  } else {
+    return (
+      <ClientSideSuspense fallback={<div />}>{() => <MultiplayerUserBar />}</ClientSideSuspense>
```

Comment:
I think it is important to wrap this in an `ErrorBoundary`, otherwise a network error can throw a top level exception.
Maybe we should even create our own `SafeLiveblocksWrapper` component which includes an ErrorBoundary and a `ClientSideSuspense`

Reply:
I absolutely agree - and I think that a dedicated component that does both things (plus more, e.g. error reporting) is the way to go. Incremental PR sounds like the best option since there's a bunch of moving pieces atm